lib/jellyfish/config_reader.ex

```diff
@@ -50,7 +50,7 @@ defmodule Jellyfish.ConfigReader do
     end
   end
 
-  def read_boolean(env) do
+  defp parse_boolean(env, other_handler \\ nil) do
```

Comment:
Let's just extend previous function i.e.

```elixir
def read_boolean(env, fallback) do
```

or modify all exisiting functions to have trailing `!` and add `read_boolean` without `!` e.g.

```elixir
def read_check_origin(env) do
  case read_boolean(env) do
    {:ok, value} -> # ...
    {:error, reason} -> # try to parse domains
  end
 end
 ```

That's how livebook does it https://github.com/livebook-dev/livebook/blob/main/lib/livebook/config.ex#L446
 
Now we have parse_boolean which also reads env variable so we mix responsibilities